### ISSUE TITLE
Establish connectivity between VRO and BIE kafka service environments

### ISSUE THREAD

**Issue by dianagriffin** [department-of-veterans-affairs/abd-vro#1674]
**User Story**

As a VRO engineer, I want to connect to BIE Kafka in my LHDI environments, so that I can develop against it.

**Context**
Completing this ticket will require technical research and coordination to answer: For each relevant BIE environment, how do we connect VRO to BIE Kafka (or what is the URL and authentication)?
- Ask @verdance-luke to demo connecting to BIE in dev #1159 
- How does DEV8 env map to LHDI envs? What are other BIE envs?
- Create a Kafka-client container image; Go to each respective LHDI envs and test connection to BIE

**Acceptance Criteria**

1. Document that each VRO environment can connect to the corresponding BIE kafka service (BIP) environment(s), and if so, how

**Notes about work**
- This ticket is part of epic #1642

**Comment by dianagriffin** [comment]
Email update from Adrien 6/21:

> Premal, hi!
> 
> With your onboarding in Dev8 environment, we initiated a CR (https://jira.devops.va.gov/browse/BPS-21301) to provision your certificates (that certificate was delivered to you for connectivity testing in Dev8).
> 
> With that CR, BIA Systems had already made certificates available for VRO in Stage8. They confirmed that ACLs were migrated to the new system.  The same certificate should work in Stage environments. Luke, can you please perform a connectivity testing in UAT for instance?
> 
> Note: Please visit the CR link for more details in comments.
> 
> Since you expressed interest in accessing Exam events in our last meeting, please use the link below to pick events that will be relevant in our next Sprint.  Feel free to reach out with any questions about the events.
> https://confluence.devops.va.gov/display/VAExternal/Domain%3A+Exams

**Comment by chengjie8** [comment]
[Connectivity_Test_Instructions_Tenant_Kafka_v1.0.docx](https://github.com/department-of-veterans-affairs/abd-vro/files/11884659/Connectivity_Test_Instructions_Tenant_Kafka_v1.0.docx)
Attached Connectivity test guide

**Comment by chengjie8** [comment]
- Followed [this guide](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Deploying-VRO) and setup lightkeeper and kube config file
- successfully deployed a sleep pod with kcat docker image to `va-abd-rrd-dev` namespace in `ldx-nonprod-1` cluster
- Split the cert/pem json file into separate crt and pem files and copied over to the pod from step 2.

So far I had no success connecting to kafka dev broker `kafka.kafkav2.dev.bip.va.gov:443`. 

Luke suggested to make sure to have official docker image based on [this guide](https://animated-carnival-57b3e7f5.pages.github.io/container-platform/signed-images/). Will connect later today with Luke and see what configuration he has.

**Comment by chengjie8** [comment]
[BIE Kafka topic JSON schema definitions](https://github.com/department-of-veterans-affairs/bip-bie)

**Comment by chengjie8** [comment]
BIE team has recently updated their kafka broker hostname which was why I had some difficult time connecting to it in the beginning. 
According to David Hamilton from BIE team

- DEV and TEST (redirect) go to `dev` cluster `kafka.dev.bip.va.gov:443`
- UAT and SQA/IVV go to `stage` cluster `kafka.stage.bip.va.gov:443`

I have the dev cluster cert/pem files and verified the connection and topics. I can connect to stage cluster, however, in order to view topics, it requires UAT specific cert/pem files.  Once Premal gets it, i will test the connection again to stage.

**Comment by chengjie8** [comment]
![image](https://github.com/department-of-veterans-affairs/abd-vro/assets/89034346/04789a8b-ca3f-417e-836b-6f1e09e1b692)

**Comment by chengjie8** [comment]
BIE team has managed to send us `ivv` and `uat` key/certs along with encryption password. I was able to verify the connection in K8s pod.

**Comment by chengjie8** [comment]
a sleeping pod has been successfully deployed in every non-production namespace, with the exception of va-abd-rrd-dev, which was confirmed previously. This includes:

- `va-abd-rrd-qa`
- `va-abd-rrd-sandbox`

In each of these pods, I have configured specific paths as follows:
- The path `/tmp/DEV` contains DEV Kafka cluster certificate/key files.
- The path `/tmp/STG` contains IVV Kafka cluster certificate/key files.
- The path `/tmp/STG-2` contains UAT Kafka cluster certificate/key files.

I have verified that both `va-abd-rrd-qa` and `va-abd-rrd-sandbox` sleeping pods can establish connections and see topics from the all `DEV/IVV/UAT` clusters. No issues with connecting to any Kafka environment, provided the appropriate certificate/key is utilized.
